common_app/main.c

```diff
@@ -0,0 +1,99 @@
+/*****************************************************************************
+ *   Ledger App Boilerplate.
+ *   (c) 2020 Ledger SAS.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *****************************************************************************/
+
+#include <stdint.h>  // uint*_t
+#include <string.h>  // memset, explicit_bzero
+
+#include "os.h"
+#include "ux.h"
+
+#include "types.h"
+#include "globals.h"
+#include "io.h"
+#include "sw.h"
+#include "ui/menu.h"
+#include "apdu/parser.h"
+#include "apdu/dispatcher.h"
+#include "macros.h"
+
+uint8_t G_io_seproxyhal_spi_buffer[IO_SEPROXYHAL_BUFFER_SIZE_B];
+ux_state_t G_ux;
+bolos_ux_params_t G_ux_params;
+
+/**
+ * Exit the application and go back to the dashboard.
+ */
+WEAK void app_exit() {
+    BEGIN_TRY_L(exit) {
+        TRY_L(exit) {
+            os_sched_exit(-1);
+        }
+        FINALLY_L(exit) {
+        }
+    }
+    END_TRY_L(exit);
+}
+
+/**
+ * Main loop to setup USB, Bluetooth, UI and launch app_main().
+ */
+WEAK __attribute__((section(".boot"))) int main() {
+    __asm volatile("cpsie i");
+
+    os_boot();
+
+    for (;;) {
+        // Reset UI
+        memset(&G_ux, 0, sizeof(G_ux));
+
+        BEGIN_TRY {
+            TRY {
+                io_seproxyhal_init();
+
+#ifdef TARGET_NANOX
+                G_io_app.plane_mode = os_setting_get(OS_SETTING_PLANEMODE, NULL, 0);
+#endif  // TARGET_NANOX
```

Comment:
Probably `HAVE_BLE` to include STAX?

Reply:
Fixed by integrating changes from boilerplate :)